plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -86,6 +86,7 @@
     private int maxConcurrentFileRenames = 20;
     private int maxConcurrentMetastoreDrops = 20;
     private int maxConcurrentMetastoreUpdates = 20;
+    private int maxPartitionDropsPerTransaction = 100_000;
```

Comment:
this seems high enough to me that it should not cause regression practically with the default conf. However, using an optional might be a little bit better to guarantee that - especially for single-tenant clusters that may not care about the bottleneck caused by this. @electrum any thoughts?